### PR TITLE
Replace CRA build utility with Vite

### PR DESCRIPTION
**Pull request by rshewitt**
- Purpose:
  - addresses [#4274](https://github.com/GSA/data.gov/issues/4274)
- Updates:
  - add Vite as build utility
    - replace react-scripts build/start scripts with Vite build/start. 
    - change all .js file extension to .jsx.
    - replace commonJS "require" statements to ESM "import"
    - add vite.config.js
  - update Node version of app to >=18.16.0 <19.0.0
  - remove some unneeded dependencies. 
  - update to latest cypress docker image. 
  - update to ckan-solr:2.10 image.

### CODE REVIEW

**Review by jbrown-xentity — APPROVED** [review]
Looks good to me, just a few tiny things!

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -38,8 +40,10 @@ jobs:
         with:
           python-version: '3.8'
       - uses: actions/setup-node@v3
+        # with:
+        #   node-version: 18.16.0
         with:
-          node-version: 16
+          node-version-file: 'package.json'
```
[the remark this answers is not in the document]
Comment (rshewitt) [reply .github/workflows/test.yml:46]: yeah I really like this configuration! I stumbled on this reading through node action documentation. I'll remove the comment.

File: docker-compose.cypress.yml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 version: "3.7"
 services:
   cypress:
-    image: cypress/included:7.7.0
+    image: cypress/included:latest
```
[the remark this answers is not in the document]
Comment (rshewitt) [reply docker-compose.cypress.yml:4]: that makes sense. I'll update this.

File: metadata-app/index.html [hunk 3]
```diff
@@ -2,25 +2,22 @@
...
-    <link rel="manifest" href="%PUBLIC_URL%/manifest.json" />
+    <link rel="manifest" href="/manifest.json" />
     <!--
-      Notice the use of %PUBLIC_URL% in the tags above.
+      Notice the use of  in the tags above.
       It will be replaced with the URL of the `public` folder during the build.
       Only files inside the `public` folder can be referenced from the HTML.
 
-      Unlike "/favicon.ico" or "favicon.ico", "%PUBLIC_URL%/favicon.ico" will
+      Unlike "/favicon.ico" or "favicon.ico", "/favicon.ico" will
```
[the remark this answers is not in the document]
Comment (rshewitt) [reply metadata-app/index.html:20]: ah yes i'll update this. thanks for pointing this out.

File: .github/workflows/test.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (rshewitt) [reply .github/workflows/test.yml]: resolved via 8f038726f3c5e1a2f8798d50d329bbcc3edcddef

File: docker-compose.cypress.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (rshewitt) [reply docker-compose.cypress.yml]: resolved via 57761f599a5337e9f7dedc49c598d098f09f65c7

File: metadata-app/index.html (hunk 3 above)
[the remark this answers is not in the document]
Comment (rshewitt) [reply metadata-app/index.html]: resolved via 842e68ca840f9e1939214dbfc3ae974e7bfed919

**Review by jbrown-xentity — DISMISSED** [review]

**Comment by rshewitt** [comment]
forgot I had black auto-format on save for the `setup.py` commit so the diff is larger than it should be. the only change was bumping the version number from `0.4.2` to `0.5.0`

**Review by jbrown-xentity — APPROVED** [review]

**Pull request merged**